src/buildercore/fastly/vcl/main.vcl

```diff
@@ -109,6 +109,12 @@ sub vcl_error {
     return(deliver_stale);
   }
 
+  if (obj.status == 302) {
+     set obj.http.Location = req.http.X-eLife-Redirect;
+     synthetic {""};
+     return (deliver);
```

Comment:
The VCL code is also executed at the shield in theory? Despite only the `Referer`-driven traffic being able to reach it. That one should also pass through the shield.

Reply:
It will, but https://github.com/elifesciences/builder/blob/313aafcacc7a4dd49efc72218ca4fb3711e535c9/src/buildercore/fastly/vcl/journal-submit.vcl.tpl#L3 means that it passes straight through (without caching).